### PR TITLE
add Combo Box docs to site

### LINKED ISSUE #2217: ComboBox Documentation

### PR DESCRIPTION
**Pull request by feliciackh**
Closes #2217

Preview: https://saltdesignsystem-git-cbb-docs-fed-team.vercel.app/salt/components/combo-box

### CODE REVIEW

File: site/docs/components/combo-box/examples.mdx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+### Default (single select)
+
+A single item can be selected from the Combo Box options.
+
+Combo Box contains an input field that opens a list. As users' type, the list displays all filtered options based on the user’s input.
+
+</LivePreview>
+
+<LivePreview componentName="combo-box" exampleName="CustomRenderer" >
+
```
Comment (feliciackh) [inline site/docs/components/combo-box/examples.mdx:24]:
```suggestion
### Default (single select)

A two item can be selected from the Combo Box options.

Combo Box contains an input field that opens a list. As users' type, the list displays all filtered options based on the user’s input.

</LivePreview>

<LivePreview componentName="combo-box" exampleName="CustomRenderer" >

```

File: site/docs/components/combo-box/examples.mdx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+<LivePreview componentName="combo-box" exampleName="Disabled" >
+
+### Disabled
+
+When `disabled={true}`, users will not be able to interact with Combo Box and it does not receive focus on any interaction.
+
+Guidance:
+
+- Disable Combo Box if you need to restrict access to options based on the current context.
```
Comment (feliciackh) [inline site/docs/components/combo-box/examples.mdx:64]:
```suggestion
- Disable combo box if you need to restrict access to options based on the current context.
```

File: site/docs/components/combo-box/examples.mdx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+data:
+  $ref: ./#/data
+---
+
+<LivePreviewControls>
+<LivePreview componentName="combo-box" exampleName="Default">
+
+### Default (single select)
+
+A single item can be selected from the Combo Box options.
```
Comment (navkaur76) [inline site/docs/components/combo-box/examples.mdx:17]:
```suggestion
A single item can be selected from the combo box options.
```

File: site/docs/components/combo-box/examples.mdx (hunk 2 above)
Comment (navkaur76) [inline site/docs/components/combo-box/examples.mdx:64]:
```suggestion
- Disable combo box if you need to restrict access to options based on the current context.
```

File: site/docs/components/combo-box/examples.mdx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+<LivePreview componentName="combo-box" exampleName="Disabled" >
+
+### Disabled
+
+When `disabled={true}`, users will not be able to interact with Combo Box and it does not receive focus on any interaction.
+
+Guidance:
+
+- Disable Combo Box if you need to restrict access to options based on the current context.
+- Showing Combo Box in a disabled state makes the user aware of the control without enabling access to its functionality.
```
Comment (navkaur76) [inline site/docs/components/combo-box/examples.mdx:65]:
```suggestion
- Showing combo box in a disabled state makes the user aware of the control without enabling access to its functionality.
```

File: packages/lab/src/combo-box-next/ComboBoxNext.tsx [hunk 5]
```diff
@@ -41,7 +41,7 @@ export interface ComboBoxNextProps<T>
   /**
    * Controlled prop. Controls the Input value in the Combo Box Input.
    */
-  inputValue: string;
+  inputValue?: string;
```
Comment (origami-z) [inline packages/lab/src/combo-box-next/ComboBoxNext.tsx:44]: This is a TS fix to the prop. Better to have a changeset

Comment (feliciackh) [reply packages/lab/src/combo-box-next/ComboBoxNext.tsx:44]: CS added

File: site/docs/components/combo-box/examples.mdx [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+<LivePreview componentName="combo-box" exampleName="Disabled" >
+
+### Disabled
+
+When `disabled={true}`, users will not be able to interact with Combo Box and it does not receive focus on any interaction.
+
+Guidance:
+
+- Disable combo box if you need to restrict access to options based on the current context.
+- Showing combo box in a disabled state makes the user aware of the control without enabling access to its functionality.
```
Comment (libertymayc) [inline site/docs/components/combo-box/examples.mdx:65]:
```suggestion
- Disable Combo Box if you need to restrict access to options based on the current context.
- Showing Combo Box in a disabled state makes the user aware of the control without enabling access to its functionality.
```

File: site/docs/components/combo-box/usage.mdx [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+- To display and select from a list of 10 or more options that don't need to be permanently visible to the user without selecting from a dropdown menu.
+- When only the selected value from a set of options needs to be visible once the selection is made.
+- If a workflow benefits from having a filter to quickly narrow down the available options.
+- If there are more than 100 options and you would like to prevent performance issues without virtualization.
+
+#### When not to use Combo Box
+
+- To choose one value from a set of five to ten options that does not require filtering. Instead, use [Dropdown](../dropdown).
+- To select a single option from fewer than five. Instead, use [Radio Button Group](../radio-button).
+- To select multiple options from fewer than five. Instead, use [Checkbox group](../checkbox).
```
Comment (libertymayc) [inline site/docs/components/combo-box/usage.mdx:25]:
```suggestion
- To select multiple options from fewer than five. Instead, use [Checkbox Group](../checkbox).
```

File: site/docs/components/combo-box/examples.mdx (hunk 6 above)
Comment (feliciackh) [reply site/docs/components/combo-box/examples.mdx:65]: will keep this as is as design's requested this change

File: site/docs/components/combo-box/examples.mdx [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+</LivePreview>
+
+<LivePreview componentName="combo-box" exampleName="CustomRenderer" >
+
+### Custom renderer
+
+Customize the list item display to visually assist users, for example rendering the flag icons next to the name of countries as seen in this example. Use the `ListItem` prop to customize list items.
+
+Guidance:
```
Comment (lilyvc) [inline site/docs/components/combo-box/examples.mdx:29]:
```suggestion
#### Guidance
```

File: site/docs/components/combo-box/examples.mdx [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+
+- Users can benefit from seeing a particular visual with list items. Customize the display as you see fit. For example, if displaying a list of contacts, it may help to provide headshots alongside the option text.
+- Place any visual support such as icons, country symbols or avatars to the left of the option text.
+
+Define a filter that only matches the first letter of a word. Use the `itemFilter` prop to customize the filter that applies to the elements.
+
+Guidance:
```
Comment (lilyvc) [inline site/docs/components/combo-box/examples.mdx:36]:
```suggestion
#### Guidance
```

File: site/docs/components/combo-box/examples.mdx [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+</LivePreview>
+
+<LivePreview componentName="combo-box" exampleName="Disabled" >
+
+### Disabled
+
+When `disabled={true}`, users will not be able to interact with Combo Box and it does not receive focus on any interaction.
+
+Guidance:
```
Comment (lilyvc) [inline site/docs/components/combo-box/examples.mdx:62]:
```suggestion
#### Guidance
```

File: site/docs/components/combo-box/examples.mdx [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+
+- Give users a quicker and easier experience when they’re choosing their options by changing the logic of the component’s typeahead filter.
+
+</LivePreview>
+
+<LivePreview componentName="combo-box" exampleName="Variants" >
+
+### Variants
+
+Combo box has two variants: `primary` and `secondary`. The primary variant is set by default. Set to secondary variant with `variants=”secondary”`.
```
Comment (lilyvc) [inline site/docs/components/combo-box/examples.mdx:46]:
```suggestion
Combo Box has two variants: `primary` and `secondary`. The primary variant is set by default. Set to secondary variant with `variants=”secondary”`.
```

File: site/docs/components/combo-box/examples.mdx [hunk 12]
```diff
@@ -0,0 +1,68 @@
...
+
+Guidance:
+
+- Use the secondary variant when in a data-dense or compact interface, or for a form with multiple columns or sections.
+- When deciding which variant to use, consider the background color of the application.
+- Tip: The [Forms pattern](/salt/patterns/forms) provides further guidance on primary versus secondary control selection. Combo box can be wrapped in a Form Field when it’s displayed within a Form context. This provides functionality built into Form Field for increased accessibility. For more information, refer to the [Form Field](../form-field) guidance.
```
Comment (lilyvc) [inline site/docs/components/combo-box/examples.mdx:52]:
```suggestion
- Tip: The [Forms pattern](/salt/patterns/forms) provides further guidance on primary versus secondary control selection. Combo Box can be wrapped in a Form Field when it’s displayed within a Form context. This provides functionality built into Form Field for increased accessibility. For more information, refer to the [Form Field](../form-field) guidance.
```

File: site/docs/components/combo-box/examples.mdx [hunk 13]
```diff
@@ -0,0 +1,68 @@
...
+
+<LivePreview componentName="combo-box" exampleName="Disabled" >
+
+### Disabled
+
+When `disabled={true}`, users will not be able to interact with Combo Box and it does not receive focus on any interaction.
+
+Guidance:
+
+- Disable combo box if you need to restrict access to options based on the current context.
```
Comment (lilyvc) [inline site/docs/components/combo-box/examples.mdx:64]:
```suggestion
- Disable Combo Box if you need to restrict access to options based on the current context.
```

File: site/docs/components/combo-box/examples.mdx (hunk 6 above)
Comment (lilyvc) [inline site/docs/components/combo-box/examples.mdx:65]:
```suggestion
- Showing Combo Box in a disabled state makes the user aware of the control without enabling access to its functionality.
```

**Comment by lilyvc** [comment]
Should you see a dropdown in the preview? I'm not sure if it is not working or not up to date

File: site/docs/components/combo-box/examples.mdx (hunk 8 above)
Comment (feliciackh) [reply site/docs/components/combo-box/examples.mdx:29]: keeping this in line with other docs from the same batch, as per design's guidance

File: site/docs/components/combo-box/examples.mdx [hunk 14]
```diff
@@ -0,0 +1,68 @@
...
+
+</LivePreview>
+
+<LivePreview componentName="combo-box" exampleName="Variants" >
+
+### Variants
+
+Combo box has two variants: `primary` and `secondary`. The primary variant is set by default. Set to secondary variant with `variants=”secondary”`.
+
+Guidance:
```
[the remark this answers is not in the document]
Comment (feliciackh) [reply site/docs/components/combo-box/examples.mdx:48]: as above

File: site/docs/components/combo-box/examples.mdx (hunk 13 above)
Comment (feliciackh) [reply site/docs/components/combo-box/examples.mdx:64]: keeping this as per design's change request

File: site/docs/components/combo-box/examples.mdx (hunk 6 above)
Comment (feliciackh) [reply site/docs/components/combo-box/examples.mdx:65]: as above

File: site/docs/components/combo-box/examples.mdx (hunk 10 above)
Comment (feliciackh) [reply site/docs/components/combo-box/examples.mdx:62]: as above

File: site/docs/components/combo-box/examples.mdx (hunk 9 above)
Comment (feliciackh) [reply site/docs/components/combo-box/examples.mdx:36]: as above

**Comment by feliciackh** [comment]
> Should you see a dropdown in the preview? I'm not sure if it is not working or not up to dat
Great catch - z index added to display list on combobox click

File: site/docs/components/combo-box/examples.mdx [hunk 15]
```diff
@@ -0,0 +1,68 @@
...
+
+- Give users a quicker and easier experience when they’re choosing their options by changing the logic of the component’s typeahead filter.
+
+</LivePreview>
+
+<LivePreview componentName="combo-box" exampleName="Variants" >
+
+### Variants
+
+Combo Box has two variants: `primary` and `secondary`. The primary variant is set by default. Set to secondary variant with `variants=”secondary”`.
```
Comment (alycrys) [inline site/docs/components/combo-box/examples.mdx:46]:
```suggestion
Combo Box has two variants: `primary` and `secondary`. The primary variant is set by default. Set to secondary variant with `variant=”secondary”`.
```

**Comment by JoanaMMoreira** [comment]
> > Should you see a dropdown in the preview? I'm not sure if it is not working or not up to dat
> > Great catch - z index added to display list on combobox click

Should the list have a blue border?

**Review by alycrys — DISMISSED** [review]

**Comment by feliciackh** [comment]
> > > Should you see a dropdown in the preview? I'm not sure if it is not working or not up to dat
> > > Great catch - z index added to display list on combobox click
> 
> Should the list have a blue border?

Very strange - locally borders are in blue, are you aware of any open issue that might have caused this

File: .changeset/pink-bees-greet.md [hunk 16]
```diff
@@ -0,0 +1,5 @@
+---
+"@salt-ds/lab": patch
+---
+
+Make combo box controlled prop inputValue optional
```
Comment (feliciackh) [inline .changeset/pink-bees-greet.md:5]:
```suggestion
- Make combo box controlled prop inputValue optional
- Add z-index to combobox list
```

**Review by lilyvc — APPROVED** [review]

**Comment by JoanaMMoreira** [comment]
> > > > Should you see a dropdown in the preview? I'm not sure if it is not working or not up to dat
> > > > Great catch - z index added to display list on combobox click
> > 
> > 
> > Should the list have a blue border?
> 
> Very strange - locally borders are in blue, are you aware of any open issue that might have caused this

I'm not sure there's an open issue related to this, could it be that that border token isn't available somehow or it's being overridden? It's tricky to inspect as it closes the list

File: packages/lab/src/combo-box-next/ComboBoxNext.css [hunk 17]
```diff
@@ -10,4 +10,5 @@
   border-color: var(--salt-selectable-borderColor-selected);
   box-shadow: var(--salt-overlayable-shadow-popout);
   max-height: calc((var(--salt-size-base) + var(--salt-spacing-100)) * var(--comboBoxNext-itemCount, 5));
+  z-index: var(--salt-zIndex-flyover);
```
Comment (JoanaMMoreira) [inline packages/lab/src/combo-box-next/ComboBoxNext.css:13]: Can we just double check this is the correct z-index token that should be used? It seems quite high

Comment (feliciackh) [reply packages/lab/src/combo-box-next/ComboBoxNext.css:13]:
```suggestion
  z-index: calc(var(--salt-zIndex-appHeader) - 1);
```

Comment (feliciackh) [reply packages/lab/src/combo-box-next/ComboBoxNext.css]: @JoanaMMoreira updated to align z-index token used with Dialog/Drawer component

**Comment by feliciackh** [comment]
> > > > > Should you see a dropdown in the preview? I'm not sure if it is not working or not up to dat
> > > > > Great catch - z index added to display list on combobox click
> > > 
> > > 
> > > Should the list have a blue border?
> > 
> > 
> > Very strange - locally borders are in blue, are you aware of any open issue that might have caused this
> 
> I'm not sure there's an open issue related to this, could it be that that border token isn't available somehow or it's being overridden? It's tricky to inspect as it closes the list

the token worked fine in several other components that referenced it, and instead of the combo box list token being applied on list, it falls back to showing list's border-color token instead. I'm only able to inspect locally by forcing its state to be active

**Review by lilyvc — DISMISSED** [review]

**Comment by JoanaMMoreira** [comment]
> > > > > > Should you see a dropdown in the preview? I'm not sure if it is not working or not up to dat
> > > > > > Great catch - z index added to display list on combobox click
> > > > 
> > > > 
> > > > Should the list have a blue border?
> > > 
> > > 
> > > Very strange - locally borders are in blue, are you aware of any open issue that might have caused this
> > 
> > 
> > I'm not sure there's an open issue related to this, could it be that that border token isn't available somehow or it's being overridden? It's tricky to inspect as it closes the list
> 
> the token worked fine in several other components that referenced it, and instead of the combo box list token being applied on list, it falls back to showing list's border-color token instead. I'm only able to inspect locally by forcing its state to be active

Changing the selector like Josh suggested should do the trick :)

File: .changeset/pink-bees-greet.md [hunk 18]
```diff
@@ -0,0 +1,6 @@
+---
+"@salt-ds/lab": patch
+---
+
+- Make combo box controlled prop inputValue optional
+- Add z-index to combobox list
```
Comment (feliciackh) [inline .changeset/pink-bees-greet.md:6]:
```suggestion
- Add z-index to combobox list
- Increase specificity for list css
```

**Review by JoanaMMoreira — APPROVED** [review]

**Review by alycrys — APPROVED** [review]

**Review by lilyvc — APPROVED** [review]

**Pull request merged**